### PR TITLE
Queue Service: Convert to Helm

### PR DESCRIPTION
**Pull request by edunham**
Need to figure out what the plan is for the s3 policy components. I pretended like all the stuff it's reading out of AWS is just hardcoded in Values; this is probably wrong but got the boilerplate done for now.

### CODE REVIEW

**Comment by edunham** [comment]
Note that it's the only service using `azure_account_id`; I assumed this was meaning `azure_account`. Could very well be a faulty assumption though.

File: infrastructure/k8s/services/queue.yaml [hunk 1]
```diff
@@ -0,0 +1,113 @@
+project_name: "taskcluster-queue"
+secrets:
+  artifact_region: ".Values.artifactRegion"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]: I think we can reuse awsRegion

File: infrastructure/k8s/services/queue.yaml [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+  aws_access_key_id: ".Values.awsAccessKeyId"
+  aws_secret_access_key: ".Values.awsSecretAccessKey"
+  azure_account_id: ".Values.tfAzureAccount"
+  azure_account_key: ".Values.azureAccountKey"
+  blob_artifact_region: ".Values.blobArtifactRegion"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]: I think we can reuse awsRegion

File: infrastructure/k8s/services/queue.yaml [hunk 3]
```diff
@@ -0,0 +1,113 @@
+project_name: "taskcluster-queue"
+secrets:
+  artifact_region: ".Values.artifactRegion"
+  aws_access_key_id: ".Values.awsAccessKeyId"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]:
```suggestion
  aws_access_key_id: ".Values.queueAwsAccessKeyId"
```

File: infrastructure/k8s/services/queue.yaml [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+  aws_secret_access_key: ".Values.awsSecretAccessKey"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]:
```suggestion
  aws_secret_access_key: ".Values.queueAwsSecretAccessKey"
```

File: infrastructure/k8s/services/queue.yaml [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+  azure_account_id: ".Values.tfAzureAccount"
+  azure_account_key: ".Values.azureAccountKey"
+  blob_artifact_region: ".Values.blobArtifactRegion"
+  force_ssl: "false"
+  monitoring_enable: "true"
+  node_env: "production"
+  private_artifact_bucket: ".Values.privateArtifactBucket"
+  private_blob_artifact_bucket: ".Values.privateBlobArtifactBucket"
+  public_artifact_bucket: ".Values.publicArtifactBucket"
+  public_blob_artifact_bucket: ".Values.publicBlobArtifactBucket"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]: For all 4 of these [buckets](https://github.com/sciurus/taskcluster/blob/cloudops/infrastructure/terraform/main.tf#L17-L102), have terraform [output](https://github.com/sciurus/taskcluster/blob/cloudops/infrastructure/terraform/outputs.tf) the bucket IDs and then read them back in as `tf` prefixed variables.

File: infrastructure/k8s/services/queue.yaml [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+  publish_metadata: "false"
+  pulse_hostname: ".Values.pulseHostname"
+  pulse_password: ".Values.pulsePassword"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]:
```suggestion
  pulse_password: ".Values.queuePulsePassword"
```

File: infrastructure/k8s/services/queue.yaml [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+  pulse_username: ".Values.pulseUsername"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]:
```suggestion
  pulse_username: ".Values.queuePulseUsername"
```

File: infrastructure/k8s/services/queue.yaml [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+  pulse_vhost: ".Values.pulseVhost"
+  taskcluster_access_token: ".Values.notifyAccessToken"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]:
```suggestion
  taskcluster_access_token: ".Values.queueAccessToken"
```

File: infrastructure/k8s/services/queue.yaml [hunk 9]
```diff
@@ -0,0 +1,113 @@
...
+  taskcluster_client_id: "static/taskcluster/purge-cache"
```
Comment (sciurus) [inline infrastructure/k8s/services/queue.yaml]:
```suggestion
  taskcluster_client_id: "static/taskcluster/queue"
```

**Review by sciurus — CHANGES_REQUESTED** [review]
First pass, just through values.

**Comment by sciurus** [comment]
The other parts look good.

File: infrastructure/terraform/to_review/queue-service.tf [hunk 10]
```diff
@@ -1,272 +0,0 @@
...
-              "${aws_s3_bucket.public_artifacts.arn}",
-              "${aws_s3_bucket.private_artifacts.arn}"
-            ]
-        }
-    ]
-}
-EOF
-}
-
-module "queue_rabbitmq_user" {
```
Comment (sciurus) [inline infrastructure/terraform/to_review/queue-service.tf:48]: We need to add this back.

**Comment by edunham** [comment]
Should be good except for needing some changes on the cloudops-infra side to properly expose those tfWhateverArtifactBucket vars.

File: infrastructure/k8s/services/queue.yaml (hunk 5 above)
Comment (edunham) [reply infrastructure/k8s/services/queue.yaml]: https://github.com/mozilla-services/cloudops-infra/pull/1105 fixes

**Comment by edunham** [comment]
https://github.com/mozilla-services/cloudops-infra/pull/1105 goes with this and covers the other side.

File: infrastructure/terraform/outputs.tf [hunk 11]
```diff
@@ -1,3 +1,19 @@
 output "azure_account" {
   value = "${azurerm_storage_account.base.name}"
-}
\ No newline at end of file
+}
+
+output "private_artifact_bucket" {
+  value = "${aws_s3_bucket.iprivate_artifacts}"
```
Comment (sciurus) [inline infrastructure/terraform/outputs.tf]:
```suggestion
  value = "${aws_s3_bucket.private_artifacts}"
```

**Review by sciurus — APPROVED** [review]
After the typo is fixed, LGTM.

**Pull request merged**